### PR TITLE
[WIP] Simplify expression checker's member access logic and fix a bug with generic typealiases

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: lib/Sema/CSGen.cpp [hunk 1]
```diff
@@ -1362,9 +1362,20 @@ namespace {
         type = E->getTypeLoc().getType();
       }
       if (!type || type->hasError()) return Type();
-      
+
```
Comment (shahzadlone) [inline lib/Sema/CSGen.cpp]: Is this your text editor just converting tabs to spaces?

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Pull request closed without merging**